### PR TITLE
chore: handle isaclprotected and isdelted in ConvertObjectToNode, update enterpriseca ingestion, add properties to ad schema

### PR DESCRIPTION
**Pull request by urangel**
## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

**Review by juggernot325 — APPROVED** [review]
This looks good to my naive eyes. It might be safer to have a second set take a look as well, though.

File: packages/go/ein/ad.go [hunk 1]
```diff
@@ -431,60 +438,122 @@ func ParseEnterpriseCAMiscData(enterpriseCA EnterpriseCA) []IngestibleRelationsh
...
+	}
+
+	return relationships
+}
+
+func handleEnterpriseCASecurity(enterpriseCA EnterpriseCA, relationships []IngestibleRelationship) []IngestibleRelationship {
+
+	if enterpriseCA.CASecurity.Collected {
+		caSecurityData := slices.Filter(enterpriseCA.CARegistryData.CASecurity.Data, func(s ACE) bool {
+			if s.RightName == "Owns" {
```
Comment (rvazarkar) [inline packages/go/ein/ad.go:492]: More const usage

File: packages/go/ein/ad.go [hunk 2]
```diff
@@ -431,60 +438,122 @@ func ParseEnterpriseCAMiscData(enterpriseCA EnterpriseCA) []IngestibleRelationsh
...
+				return false
+			} else {
+				return true
+			}
+		})
+
+		filteredACES := slices.Filter(enterpriseCA.Aces, func(s ACE) bool {
+			if s.RightName == "ManageCA" || s.RightName == "MangeCertificates" || s.RightName == "Enroll" {
```
Comment (rvazarkar) [inline packages/go/ein/ad.go:500]: More const usage

File: cmd/api/src/daemons/datapipe/convertors.go [hunk 3]
```diff
@@ -208,8 +212,12 @@ func convertEnterpriseCAData(data []ein.EnterpriseCA) ConvertedData {
 	converted := ConvertedData{}
 
 	for _, enterpriseca := range data {
+
+		enterpriseca.Properties[ad.CASecurityCollected.String()] = enterpriseca.CASecurity.Collected
```
[the remark this answers is not in the document]
Comment (urangel) [reply cmd/api/src/daemons/datapipe/convertors.go]: done, here is the commit:
https://github.com/BloodHoundAD/SharpHound/commit/1ce653e208a28b6f10ac5acf91525ac258c3a7a6

File: packages/cue/bh/ad/ad.cue [hunk 4]
```diff
@@ -573,6 +678,71 @@ WritePKINameFlag: types.#Kind & {
...
+	schema: "active_directory"
+}
+
+GoldenCert: types.#Kind & {
+	symbol: "GoldenCert"
+	schema: "active_directory"
+}
+
+EnrollOnBehalfOF: types.#Kind & {
+	symbol: "EnrollOnBehalfOF"
```
Comment (rvazarkar) [inline packages/cue/bh/ad/ad.cue:707]: Is the OF intended to be all caps?

File: packages/go/ein/ad.go [hunk 5]
```diff
@@ -34,9 +35,13 @@ func ConvertSessionObject(session Session) IngestibleSession {
 }
 
 func ConvertObjectToNode(item IngestBase, itemType graph.Kind) IngestibleNode {
+	properties := item.Properties
+	properties[ad.IsACLProtected.String()] = item.IsACLProtected
```
Comment (rvazarkar) [inline packages/go/ein/ad.go:39]: This is probably ok, but I would still prefer to add these properties during collection.

File: packages/go/ein/ad.go [hunk 6]
```diff
@@ -431,60 +438,122 @@ func ParseEnterpriseCAMiscData(enterpriseCA EnterpriseCA) []IngestibleRelationsh
...
+	relationships = handleEnterpriseCAEnrollmentAgentRestrictions(enterpriseCA, relationships, enabledCertTemplates)
+	relationships = handleEnterpriseCASecurity(enterpriseCA, relationships)
+
+	return relationships
+}
+
+func handleEnterpriseCAEnrollmentAgentRestrictions(enterpriseCA EnterpriseCA, relationships []IngestibleRelationship, enabledCertTemplates []string) []IngestibleRelationship {
+
+	if enterpriseCA.CARegistryData.EnrollmentAgentRestrictions.Collected {
+		for _, restiction := range enterpriseCA.CARegistryData.EnrollmentAgentRestrictions.Restrictions {
```
Comment (rvazarkar) [inline packages/go/ein/ad.go:461]: Typo here in restriction

File: packages/cue/bh/ad/ad.cue (hunk 4 above)
Comment (urangel) [reply packages/cue/bh/ad/ad.cue]: no... 🙃 Fixed it

File: packages/go/ein/ad.go (hunk 6 above)
Comment (urangel) [reply packages/go/ein/ad.go]: Thanks for catching these typos! Fixed

File: packages/go/ein/ad.go (hunk 5 above)
Comment (urangel) [reply packages/go/ein/ad.go]: Ripped this out 👍

**Pull request merged**